### PR TITLE
TDI-36627: fix the component part and fix a refresh issue

### PR DESCRIPTION
**Pull request by wwang-talend**
https://jira.talendforge.org/browse/TDI-36627

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/17/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/17/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/19/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/19/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/22/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/22/)

File: components-dataprep/src/test/java/org/talend/components/dataprep/connection/DataPrepServerMock.java [hunk 1]
```diff
@@ -105,12 +111,16 @@ public ResponseEntity create(@RequestHeader(value = "Authorization") String auth
...
     @RequestMapping(value = "/api/datasets/{id}", method = RequestMethod.PUT)
-    public ResponseEntity update(@RequestHeader(value = "Authorization") String authorization, @PathVariable String id,
-            InputStream inputStream) throws IOException {
+    public ResponseEntity update(@RequestHeader(value = "Authorization") String authorization,
+            @RequestHeader(value = "Accept") String accept, @PathVariable String id, InputStream inputStream) throws IOException {
         if (!TOKEN.equals(authorization)) {
             return new ResponseEntity(HttpStatus.UNAUTHORIZED);
         }
 
+        if (!ACCEPT.equals(accept)) {
```
Comment (sgandon) [inline components-dataprep/src/test/java/org/talend/components/dataprep/connection/DataPrepServerMock.java]: I don't understand why you updated the mock server and there is not unit test changed after this modification. What is the reason please ?

Comment (wwang-talend) [inline components-dataprep/src/test/java/org/talend/components/dataprep/connection/DataPrepServerMock.java]: will set the "Accept" header for some rest request, and it's necessary, so add the header check.
urlConnection.setRequestProperty("Accept", "application/json, text/plain");

**Pull request merged**